src/main/java/io/cdap/plugin/gcp/dataplex/common/util/DataplexUtil.java

```diff
@@ -0,0 +1,414 @@
+            Job currentJob =
+              dataplexServiceClient.getJob(JobName.newBuilder().setProject(projectID).setLocation(location)
+                .setLake(lake).setTask(taskId).setJob(dataplexJob.getUid()).build());
+            LOG.debug("Job is still in running state");
+            return currentJob.getState() != null && !Job.State.RUNNING.equals(currentJob.getState());
+          });
+      } catch (ConditionTimeoutException e) {
+        throw new IOException("Job timed out.", e);
+      }
+      Job completedJob = dataplexServiceClient.getJob(JobName.newBuilder().setProject(projectID).setLocation(location)
+        .setLake(lake).setTask(taskId).setJob(dataplexJob.getUid()).build());
+      if (!Job.State.SUCCEEDED.equals(completedJob.getState())) {
+        throw new IOException("Job failed with message: " + completedJob.getMessage());
+      }
+    // Default input dir path
+    String outputLocation = conf.get(INPUT_DIR);
+    // Output location path for dataplex job will be in format
+    // 'projects/projectId/locations/locationName/lakes/lakeName/tasks/taskName/jobId/0/'
+    // Need to remove 'jobs/' and add '/0/' to get the actual location
+    outputLocation = outputLocation + completedJob.getName().replace("/jobs/", "/") + "/0/";
+    conf.set(INPUT_DIR, outputLocation);
+    }
+  }
+
+  private static GoogleCredentials getCredentialsFromConfiguration(Configuration configuration) throws IOException {
+    String serviceAccount;
+    String serviceAccountType = configuration.get(DataplexConstants.SERVICE_ACCOUNT_TYPE);
+    Boolean isServiceAccountJson = false;
+    if (serviceAccountType == GCPConnectorConfig.SERVICE_ACCOUNT_JSON) {
+      isServiceAccountJson = true;
+      serviceAccount = configuration.get(SERVICE_ACCOUNT_JSON);
+    } else {
+      serviceAccount = configuration.get(SERVICE_ACCOUNT_KEYFILE);
+    }
+    String filePath = configuration.get(DataplexConstants.SERVICE_ACCOUNT_FILEPATH);
+    return getCredentialsFromServiceAccount(isServiceAccountJson, filePath, serviceAccount);
+  }
+
+  private static GoogleCredentials getCredentialsFromServiceAccount(Boolean isServiceAccountJson, String filePath,
+                                                                    String serviceAccount) throws IOException {
+    if (isServiceAccountJson || (filePath != null && !filePath.equalsIgnoreCase(DataplexConstants.NONE))) {
+      return GCPUtils.loadServiceAccountCredentials(serviceAccount, !isServiceAccountJson)
+        .createScoped(Lists.newArrayList(DataplexConstants.DATAPLEX_SCOPE));
+    } else {
+      return ServiceAccountCredentials.getApplicationDefault().createScoped(
+        Lists.newArrayList(DataplexConstants.DATAPLEX_SCOPE));
+    }
+  }
+
+  /**
+   * Return DataplexServiceClient to call dataplex methods.
+   *
+   * @param credentials GoogleCredentials
+   * @return DataplexServiceClient
+   */
+  public static DataplexServiceClient getDataplexServiceClient(GoogleCredentials credentials) {
```

Comment:
If this can return null it should be marked as `@Nullable`. Same with any other methods

Reply:
instead will throw an exception as suggested by Prashant.